src/main/java/com/fundynamic/d2tm/game/math/Vector2D.java

```diff
@@ -1,35 +1,46 @@
 package com.fundynamic.d2tm.game.math;
 
-public class Vector2D {
+public class Vector2D<T extends Number> {
 
-    private final int x, y;
+    private final T x, y;
 
-    public Vector2D(int x, int y) {
+    public static Vector2D zero() {
+        return new Vector2D(0, 0);
+    }
+
+    public Vector2D(T x, T y) {
         this.x = x;
         this.y = y;
     }
 
-    public int getX() {
+    public T getX() {
         return x;
     }
 
-    public int getY() {
+    public T getY() {
         return y;
     }
 
-    public Vector2D moveUp() {
-        return new Vector2D(x, y - 1);
+    public Vector2D<Float> move(float xVelocity, float yVelocity, float speed) {
```

Comment:
I don't like the speed thingy here. I think a vector should have a size based on the values of the x and y velocity. It shouldn't need a separate component to express that.

I.e. how is a `Vector2D(3, 4, 5)` different from a `Vector2D(15, 20, 1)`?